Получить все данные рекомендации

Leilorr opened the pull request:
Условия задачи
Реализовать возможность получения всех рекомендаций, которые пользователь дал другим пользователям.

Leilorr committed:
Получить все данные рекомендации
Условия задачи
Реализовать возможность получения всех рекомендаций, которые пользователь дал другим пользователям.

Leilorr committed:
Получить все данные рекомендации
Условия задачи
Реализовать возможность получения всех рекомендаций, которые пользователь дал другим пользователям.

Leilorr committed:
Merge branch 'feature-BJS2-1401' into feature-BJS2-1461

mirayoo commented on src/main/java/school/faang/user_service/controller/RecommendationController.java:
```diff
@@ -5,15 +5,20 @@
     private final RecommendationService recommendationService;
 
+
+    @GetMapping("/recommendations/user/{authorId}")
```
Смотри, в файлах проекта есть такой `FEATURES.md`. Там даны эндпоинты запросов
Тут должен быть такой эндпоинт: `/recommendation/given/{authorId}`

mirayoo commented on src/main/java/school/faang/user_service/controller/RecommendationController.java:
```diff
@@ -5,15 +5,20 @@
+    public List<RecommendationDto> getAllGivenRecommendations(@PathVariable long authorId) {
+        return recommendationService.getAllGivenRecommendations(authorId);
+
     @GetMapping("/recommendations/user/{receiverId}")
```
Тоже по поводу энтпоинта. Должен быть такой: `/recommendation/received/{receiverId}`

mirayoo commented on src/main/java/school/faang/user_service/service/RecommendationService.java:
```diff
@@ -8,17 +8,24 @@
+                .stream()
+                .map(recommendationMapper::toDto)
+                .toList();
+    }
     public List<RecommendationDto> getAllUserRecommendations(long receiverId){
```
Это субъективщина, просто мне кажется, что лучше разделять методы пустой строкой)

mirayoo commented on src/test/java/school/faang/user_service/controller/RecommendationControllerTest.java:
```diff
@@ -1,8 +1,11 @@
 package school.faang.user_service.controller;
 
+
```
Тут убрать лишнюю строку)

mirayoo commented on src/test/java/school/faang/user_service/controller/RecommendationControllerTest.java:
```diff
@@ -26,13 +30,20 @@ class RecommendationControllerTest {
     private RecommendationController recommendationController;
-
+  
+  
```
Убрать пустую строку

mirayoo commented on src/test/java/school/faang/user_service/controller/RecommendationControllerTest.java:
```diff
@@ -26,13 +30,20 @@ class RecommendationControllerTest {
     }
 
+    @Test
+    public void testMethod() {
```
В нейминге не силен, думаю стоит погуглить, как именовать методы тестирования

mirayoo commented on src/test/java/school/faang/user_service/service/RecommendationServiceTest.java:
```diff
@@ -19,22 +21,33 @@
     @Mock
     private RecommendationRepository recommendationRepository;
-
     @Mock
     private RecommendationMapper recommendationMapper;
```
Думаю, стоит убедиться в том, что маппер работает, а не возвращает null (через debug)
Если вдруг окажется что не работает - можно использовать класс `RecommendationMapperImpl` (генерируется при сборке приложения) и вместо `@Mock` использовать `@Spy`

Либо через `mappers.getMapper` - Влад приводил пример на стриме по юнит-тестам

mirayoo commented on src/test/java/school/faang/user_service/controller/RecommendationControllerTest.java:
```diff
@@ -26,13 +30,20 @@ class RecommendationControllerTest {
 
+    @Test
+    public void testMethod() {
+    recommendationController.getAllGivenRecommendations(1L);
+        verify(recommendationService, times(1)).getAllGivenRecommendations(1L);
+    }
```
Тесты для контроллера пока не писал, не знаю как там правильно
Но как будто не хватает `assertEquals`, чтобы убедиться в правильности отработки метода :D
Думаю, стоит у Инны уточнить этот момент)

mirayoo reviewed (commented):
Оставил пару комментов, там всякие мелочи в основном

Leilorr committed:
RecommendationController.java

Leilorr committed:
RecommendationController.java

Leilorr committed:
RecommendationService.java

Leilorr committed:
Update RecommendationControllerTest.java

Leilorr committed:
Update RecommendationDto.java

Leilorr committed:
RecommendationService.java

Leilorr committed:
RecommendationControllerTest.java

Leilorr committed:
RecommendationServiceTest.java

InnMish commented on src/main/java/school/faang/user_service/controller/RecommendationController.java:
```diff
@@ -5,17 +5,16 @@
-    @GetMapping("/recommendations/user/{receiverId}")
-    public List<RecommendationDto> getAllUserRecommendations(@PathVariable long receiverId){
-        return recommendationService.getAllUserRecommendations(receiverId);
-    }
+    @GetMapping("/recommendation/given/{authorId}")
+    public List<RecommendationDto> getAllGivenRecommendations(@PathVariable long authorId) {
+        return recommendationService.getAllGivenRecommendations(authorId);
 }
```
Ты только не удаляй то, что уже в ветке 1401 
Просто продолжай дальше работать, сохраняя старые метод 
А то потом конфликтов куча будет

InnMish commented on src/test/java/school/faang/user_service/service/RecommendationServiceTest.java:
```diff
@@ -19,43 +21,28 @@
-
     @InjectMocks
     private RecommendationService recommendationService;
+    private Page<Recommendation> page;
```
А это откуда? Мы такое еще не проходили) Точно тебе Page нужно?

InnMish commented on src/test/java/school/faang/user_service/service/RecommendationServiceTest.java:
```diff
@@ -19,43 +21,28 @@
+    @BeforeEach
+    public void init() {
+        Recommendation recommendation = new Recommendation();
+        page = new PageImpl<>(List.of(recommendation));
```
и тут тоже)

InnMish commented on src/main/java/school/faang/user_service/service/RecommendationService.java:
```diff
@@ -8,17 +8,18 @@
-        return recommendationRepository.findAllByReceiverId(receiverId, Pageable.unpaged())
+
+    public List<RecommendationDto> getAllGivenRecommendations(long authorId) {
+        return recommendationRepository.findAllByAuthorId(authorId, Pageable.unpaged())
```
Page точно нужно?

InnMish committed:
deleted  paging

Leilorr committed:
Получить все данные рекомендации
Условия задачи
Реализовать возможность получения всех рекомендаций, которые пользователь дал другим пользователям.

InnMish commented on src/main/java/school/faang/user_service/service/RecommendationService.java:
```diff
@@ -8,17 +8,18 @@
+    public List<RecommendationDto> getAllGivenRecommendations(long authorId) {
+        return recommendationRepository.findAllByAuthorId(authorId)
                 .stream()
                 .map(recommendationMapper::toDto)
```
нужно написать тест на маппер и проверить, что он работает 
Потому что я уверена, что сейчас не работает

Leilorr committed:
Условия задачи
Реализовать возможность получения всех рекомендаций, которые пользователь дал другим пользователям.

InnMish committed:
Merge remote-tracking branch 'origin/feature-BJS2-1401' into feature-BJS2-1461

# Conflicts:
#	src/main/java/school/faang/user_service/dto/recommendation/RecommendationDto.java
#	src/main/java/school/faang/user_service/mapper/RecommendationMapper.java
#	src/main/java/school/faang/user_service/service/RecommendationService.java
#	src/test/java/school/faang/user_service/controller/RecommendationControllerTest.java
#	src/test/java/school/faang/user_service/service/RecommendationServiceTest.java

InnMish committed:
refactor
